## Определение скорости полета пули при помощи баллистического маятника

**Цель**: Определить скорость полёта пули применяя законы сохранения и используя баллистические маятники.

Оборудование: духовое ружьё на штативе, осветитель, оптическая система для измерения отклонений маятника, измерительная линейка, пули и весы для их взвешивания, баллистические маятники.

### 0 Введение

#### Теория

##### Баллистический маятник, совершающий поступательное движение

В первой части работы будем использовать установку, изображённую на рисунке ниже. При попадании пули в цилиндр любая его точка движется по окружности известного радиуса, поэтому его смещение с помощью собирающей линзы можно перевести в линейное отклонение на линейке.

<img src="image_1.png" alt="Баллистический маятник, совершающий поступательное движение" style="width:600px">

При контакте пули с цилиндром можно записать ЗСИ:

$$
mu = (M+m)V
$$

где $m$ -- масса пули, $u$ -- скорость пули перед ударом, $V$-скорость цилиндра вместе с пулей после удара.

Тогда скорость пули можно выразить как:

$$
u=\frac{M}{m} \sqrt{\frac{g}{L}} \Delta x
$$

Затуханием колебанием в данном эксперименте можно пренебречь.

##### Крутильный баллистический маятник

Во второй части работы мы будем использовать крутильный баллистический маятник. Схема установки представлена на картинке ниже.

<img src="image_2.png" alt="Баллистический маятник, совершающий поступательное движение" style="width:600px">

Считая удар неупругим, можно записать уравнение
$$mur=I \Omega$$

где $r-$расстояние от линии полёта пули до оси вращения, $I$ — момент инерции относительно этой оси, $\Omega$ — угловая скорость маятника сразу после удара.

Можно пренебречь затуханием колебаний и потерями энергии и записать ЗСЭ:
$$ k \frac{\varphi^2}{2} = I \frac{\Omega^2}{2} $$

где $k$ — модуль кручения проволоки, $\varphi$ — максимальный угол поворота маятника, тогда скорость:

$$
u = \varphi \frac{\sqrt{kI}}{mr} 
$$

**Импорт библиотек и данных**

In [1]:
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns 

import statistics as st
import math as mt

### 1 Баллистический маятник, совершающий поступательное движение

In [4]:
m = np.array([0.514, 0.514, 0.509, 0.506, 0.507])
M = np.array([2925.0, 732.05])
L = 221.00
r = 22
R = 34
d = 135.8

In [6]:
A = np.array([1.18, 1.2, 1.1, 0.94, 1.12])

v_1 = np.sqrt(981.5 / L) * ((M[0] + m) / m) * A / 100

print(v_1)

[141.53703196 143.93596471 133.23715677 114.53213198 136.19470398]


Относительная погрешность:

$$
\epsilon_v^2 = \epsilon_m^2 + \epsilon_M^2 + \frac{1}{4} \epsilon_L^2 + \epsilon_A^2
$$

Абсолютная погрешность:

$$
\Delta v = \epsilon_v * v
$$

In [8]:
epsilon_M_1 = 5 / 2925

epsilon_m = 0.001 / m
epsilon_L = 1 / L
epsilon_A = 0.05 / A

epsilon_v_1 = np.sqrt(epsilon_M_1**2 + epsilon_m**2 + 0.25 * epsilon_L**2 + epsilon_A**2)
delta_v_1 = v_1 * epsilon_v_1

In [11]:
V_1 = np.average(v_1)
Delta_v_1 = np.average(delta_v_1)
Epsilon_v_1 = np.average(epsilon_v_1)
print(f"v_1 ={V_1: .2f} +-{Delta_v_1: .2f}, epsilon ={Epsilon_v_1*100: .2f} %")

v_1 = 133.89 +- 6.06, epsilon = 4.56 %


### 2 Крутильный баллистический маятник

Найдем момент инерции маятника и углы отклонения  $\varphi$, учитывая что $\varphi \approx \frac{x}{d}$:

In [16]:
m = np.array([0.511, 0.51, 0.504, 0.512, 0.511])

x = np.array([3.65, 4.05, 3.8, 4.0, 4.05])
T2 = 24.53 / 5
T1 = 33.13 / 5
kI_2 = (4 * np.pi * M[1] * R**2 * T1) / (T1**2 - T2**2)

phi = x / d
print(kI_2)

3552440.3475030893


Вычислим скорости пуль по формуле $v = \varphi \frac{\sqrt{kI}}{mr}$:

In [17]:
v_2 = phi * kI_2 / (r * m) / 100
print(v_2)

[84.93296931 94.42547764 89.6514676  92.89543518 94.24069197]


Найдем погрешности:

In [19]:
epsilon_m = 0.001 / m
epsilon_kI_2 = (0.5/M[1])**2 + (0.5/R)**2 + 9/4 * (0.5/T1)**2 + 4*(0.5/T2)**2
epsilon_r = 0.5/r
epsilon_phi = (0.5/x)**2 + (0.5/d)**2
epsilon_v_2 = np.sqrt(epsilon_kI_2 + epsilon_r**2 + epsilon_m**2 + epsilon_phi)

delta_v_2 = v_2 * epsilon_v_2

print(delta_v_2)
print(epsilon_v_2)

[23.0848175  25.04533589 24.12663295 24.7065555  24.99632082]
[0.27180043 0.26523918 0.26911587 0.26596092 0.26523915]


In [21]:
V_2 = np.average(v_2)
Delta_v_2 = np.average(delta_v_2)
Epsilon_v_2 = np.average(epsilon_v_2)
print(f"v_2 = {V_2: .2f} +-{Delta_v_2: .2f}, epsilon ={Epsilon_v_2*100: .2f} %")

v_2 =  91.23 +- 24.39, epsilon = 26.75 %


### 3 Выводы

0) В ходе работы были исследованы скорости пуль, выпущенных из 2х различных пневматических ружей с использованием поступательного и крутильного баллистического маятников.

1) Скорость пули из 1 ружья составила $ 133.89 \pm 6.06, \varepsilon_v = 4.56% $, результат точный.

2) Скорость пули из 2 ружья составила $ 91.23 \pm 24.39, \varepsilon_v = 26.75% $, результат не точный, наибольшую погрешность внесло измерение момента инерции установки.